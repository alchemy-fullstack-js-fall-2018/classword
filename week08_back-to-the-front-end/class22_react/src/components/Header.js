import React, { Component } from 'react';
import PropTypes from 'prop-types';

import alchemy from '../assets/alchemy.png';

export default class Header extends Component {
  static propTypes = {
    title: PropTypes.string
  };

  render() {
    return (
      <header>
        <img src={alchemy} />
        <h1>{this.props.title || 'Default Title'}</h1>
      </header>

    );
  }
}
