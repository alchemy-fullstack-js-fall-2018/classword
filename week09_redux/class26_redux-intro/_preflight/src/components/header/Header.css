@import "../../styles/variables.css";

.header {
  nav {
    display: flex;
    justify-content: space-evenly;
    background-color: var(--primary);
    font-size: 1.5em;

    a {
      text-decoration: none;
      padding: 0.5em 1em;
      text-align: center;
      background-color: var(--primary);
      color: var(--lightWhite);

      &:visited {
        background-color: var(--primary);
        color: var(--lightWhite);
      }

      &:hover {
        background-color: var(--lightWhite);
        color: var(--primary);
      }

    }
  }
}
