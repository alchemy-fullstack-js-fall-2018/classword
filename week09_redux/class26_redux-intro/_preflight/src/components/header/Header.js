import React from 'react';
import { Link } from 'react-router-dom';
import { ROUTES } from '../../routes/routes';

export default function Header() {
  return (
    <header>
      <nav>
        <Link to={ROUTES.HOME.linkTo()}>Home</Link>
        <Link to={ROUTES.HI_NOTES.linkTo()}>Hi Notes</Link>
      </nav>
      <h1>Note taking</h1>
    </header>
  );
}
