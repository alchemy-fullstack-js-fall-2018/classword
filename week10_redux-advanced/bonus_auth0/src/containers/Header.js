import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { getUser } from '../selectors/session';
import { ROUTES } from '../routes';

const Header = ({ user }) => {
  return (
    <header>
      <nav>
        <Link to={ROUTES.HOME.linkTo()}>HOME</Link>
        {!user && <Link to={ROUTES.LOGIN.linkTo()}>Login</Link>}
        {!user && <Link to={ROUTES.SIGNUP.linkTo()}>Signup</Link>}
        {user && <Link to={ROUTES.ARTISTS.linkTo()}>Artists</Link>}
      </nav>
      {user && <img src={(user.user_metadata && user.user_metadata.photo) || user.picture} />}
    </header>
  );
};

const mapStateToProps = state => ({
  user: getUser(state)
});

export default connect(mapStateToProps)(Header);
