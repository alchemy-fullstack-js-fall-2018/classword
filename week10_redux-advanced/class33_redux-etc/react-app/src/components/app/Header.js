import React from 'react';
import { rootLinks } from '../../routes';
import styles from './Header.module.css';

export default function Header() {
  console.log(styles)
  return (
    <header className={styles.header}>
      <nav>
        {rootLinks}
      </nav>
    </header>
  )
}
