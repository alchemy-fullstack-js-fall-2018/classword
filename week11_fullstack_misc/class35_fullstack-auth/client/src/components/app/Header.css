@import "../../css/variables.css";

.header {

}

.nav {
  display: flex;
  justify-content: space-around;
  a {
    padding: 1em;
    background-color: $primary;
    color: $text-light;
  }
}
