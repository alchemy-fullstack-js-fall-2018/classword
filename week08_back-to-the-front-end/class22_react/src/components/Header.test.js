import React from 'react';
import Header from './Header';
import render from 'react-test-renderer';

describe('Header', () => {
  test('header snapshot', () => {
    expect(render
      .create(<Header title="my title" />)
      .toJSON()).toMatchSnapshot();
  });

  test('header snapshot with default title', () => {
    expect(render
      .create(<Header />)
      .toJSON()).toMatchSnapshot();
  });
});
