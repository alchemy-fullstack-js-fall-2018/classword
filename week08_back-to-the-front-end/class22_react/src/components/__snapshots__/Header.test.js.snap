// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header header snapshot 1`] = `
<header>
  <img
    src="/path/image"
  />
  <h1>
    my title
  </h1>
</header>
`;

exports[`Header header snapshot with default title 1`] = `
<header>
  <img
    src="/path/image"
  />
  <h1>
    Default Title
  </h1>
</header>
`;
